packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -234,7 +234,7 @@ export default class ChildProcessWorker implements WorkerInterface {
       return onProcessEnd(...args);
     };
 
-    this._onCustomMessage = (...arg) => onCustomMessage(...arg);
+    this._onCustomMessage = onCustomMessage;
```

Comment:
What's the reason for this change? 

This change will have an implicit side effect of changing the `this` context value inside the `onCustomMessage` callback function.

In the old way, `this` inside the callback was whatever it was in the context where it was created (e.g. the parent object or class it was defined in), but this way it'll become the current instance of `ChildProcessWorker`.

This is probably not what you want and could cause subtle bugs.

Reply:
I checked the references for this `send` function, they all were passing in `onCustomMessage` as anonymous functions. So, the `this` instance should be passed further down.

The `(...arg)` seemed ugly. If it's not the case above, I'll revert it back

Reply:
I don't think either is a big deal here, so I'm happy with either implementation. In terms of API design, I'm a big fan of the principle of least surprise -- meaning, do the thing that the consumer of the API would expect.

In this case, if the developer passes a method of some class as the callback to this, they'd probably expect the `this` variable to remain bound to that class. I can't think of a case where they'd want the opposite?